ts/features/idpay/initiative/timeline/components/TimelineRefundDetailsComponent.tsx

```diff
@@ -0,0 +1,171 @@
+import { IOBadge } from "../../../../../components/core/IOBadge";
+import { HSpacer, VSpacer } from "../../../../../components/core/spacer/Spacer";
+import { Body } from "../../../../../components/core/typography/Body";
+import { IOStyles } from "../../../../../components/core/variables/IOStyles";
+import I18n from "../../../../../i18n";
+import NavigationService from "../../../../../navigation/NavigationService";
+import { useIOSelector } from "../../../../../store/hooks";
+import themeVariables from "../../../../../theme/variables";
+import { formatNumberAmount } from "../../../../../utils/stringBuilder";
+import { IDPayConfigurationRoutes } from "../../configuration/navigation/navigator";
+import { idpayInitiativeIdSelector } from "../../details/store";
+
+type Props = {
+  refund: RefundDetailDTO;
+};
+
+const TimelineRefundDetailsComponent = (props: Props) => {
+  const { refund } = props;
+
+  const { close: closeBottomSheet } = useBottomSheet();
+  const initiativeId = useIOSelector(idpayInitiativeIdSelector);
+
+  const handleEditIbanPress = () => {
+    closeBottomSheet();
+
+    // The utilization of the NavigationService is imperative, as the library employed for presenting bottom sheets
+    // utilizes its own NavigationContainer. Consequently, utilizing the "useNavigation" hook in this scenario
+    // would result in an error.
+    NavigationService.dispatchNavigationAction(
+      CommonActions.navigate(
+        IDPayConfigurationRoutes.IDPAY_CONFIGURATION_MAIN,
+        {
+          screen: IDPayConfigurationRoutes.IDPAY_CONFIGURATION_IBAN_ENROLLMENT,
+          params: {
+            initiativeId
+          }
+        }
+      )
+    );
+  };
+
+  const alertViewRef = React.createRef<View>();
+
+  const rejectedAlertComponent = pipe(
+    refund.operationType,
+    O.of,
+    O.filter(type => type === OperationTypeEnum.REJECTED_REFUND),
+    O.map(() => (
+      <>
+        <Alert
+          viewRef={alertViewRef}
+          content={I18n.t(
+            "idpay.initiative.operationDetails.refund.rejectedAdvice.text"
+          )}
+          action={I18n.t(
+            "idpay.initiative.operationDetails.refund.rejectedAdvice.editIban"
+          )}
+          onPress={handleEditIbanPress}
+          variant="error"
+        />
+        <VSpacer size={16} />
+      </>
+    )),
+    O.toNullable
+  );
+
+  const refundAmount = pipe(
+    refund.amount,
+    O.fromNullable,
+    O.alt(() => O.some(0)),
+    O.map(amount => formatNumberAmount(amount, true)),
+    O.toNullable
+  );
+
+  const periodDateString = pipe(
+    sequenceS(O.Monad)({
+      startDate: pipe(
+        refund.startDate,
+        O.fromNullable,
+        O.map(date => format(date, "DD/MM/YY"))
+      ),
+      endDate: pipe(
+        refund.endDate,
+        O.fromNullable,
+        O.map(date => format(date, "DD/MM/YY"))
+      )
+    }),
+    O.map(({ startDate, endDate }) => `${startDate} - ${endDate}`),
+    O.getOrElse(() => "-")
+  );
```

Comment:
Could make sense to have this type of function in a separate file? It might facilitate testability as well. What do you think?

Reply:
Yes this is a great idea!
In https://github.com/pagopa/io-app/pull/4370/commits/4b83d4ae1909b21609562c61f8c66b609917dc3b, I have moved `periodDateString` in its own file and I have added tests as well.
Also, I have removed `refundAmount` pipe since `refund.amount` is no more a nullable value, so we can use `formatNumberAmount(amount, true)` directly in the component tree.